projects/igniteui-angular/src/lib/grids/row-drag.directive.ts

```diff
@@ -0,0 +1,137 @@
+import { Directive, Input, OnDestroy, NgModule } from '@angular/core';
+import { IgxDragDirective } from '../directives/dragdrop/dragdrop.directive';
+import { IRowDragEndEventArgs, IRowDragStartEventArgs } from './grid-base.component';
+import { KEYS } from '../core/utils';
+import { fromEvent, Subscription } from 'rxjs';
+import { IgxRowComponent, IgxGridBaseComponent, IGridDataBindable } from './grid';
+
+
+const ghostBackgroundClass = 'igx-grid__tr--ghost';
+const gridCellClass = 'igx-grid__td';
+
+/**
+ * @hidden
+ */
+@Directive({
+    selector: '[igxRowDrag]'
+})
+export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
+    private row: IgxRowComponent<IgxGridBaseComponent & IGridDataBindable>;
+    private subscription$: Subscription;
+    private _rowDragStarted = false;
+    public startDrag(event) {
+        this.onPointerDown(event);
+    }
+
+    @Input('igxRowDrag')
+    set data(val) {
+        this.row = val;
+    }
+
+    get data() {
+        return this.row;
+    }
+
+    public onPointerDown(event) {
+        event.preventDefault();
+        this._rowDragStarted = false;
+        super.onPointerDown(event);
+    }
+
+    public onPointerMove(event) {
+        super.onPointerMove(event);
+        if (this._dragStarted && !this._rowDragStarted) {
+            this._rowDragStarted = true;
+            const args: IRowDragStartEventArgs = {
+                owner: this,
+                dragData: this.row,
+                cancel: false
+            };
+
+            this.row.grid.onRowDragStart.emit(args);
+            if (args.cancel) {
+                this._clicked = false;
+                return;
+            }
+            this.row.dragging = true;
+            this.row.grid.rowDragging = true;
+            if (this.row.grid.rowEditable && this.row.grid.rowInEditMode) {
+                this.row.grid.endEdit(true);
+            }
+            this.row.grid.markForCheck();
+
+            this.subscription$ = fromEvent(this.row.grid.document.defaultView, 'keydown').subscribe((ev: KeyboardEvent) => {
+                if (ev.key === KEYS.ESCAPE || ev.key === KEYS.ESCAPE_IE) {
```

Comment:
I would assume escape handling to be part of the base drag directive, then again is it really needed, considering you can just release the the pointer?